pycloudlib/gce/cloud.py

```diff
@@ -153,36 +173,77 @@ def launch(self, image_id, instance_type='n1-standard-1', user_data=None,
             },
         }
 
+        if user_data:
+            user_metadata = {
+                'key': 'user-data',
+                'value': user_data
+            }
+            config['metadata']['items'].append(user_metadata)
+
         operation = self.compute.instances().insert(
             project=self.project,
             zone=self.zone,
             body=config
         ).execute()
+        raise_on_error(operation)
 
-        self._wait_for_operation(operation)
+        self._wait_for_operation(operation, operation_type='zone')
 
         result = self.compute.instances().get(
             project=self.project,
             zone=self.zone,
-            instance=self.tag
+            instance=instance_name,
         ).execute()
+        raise_on_error(result)
 
         self._log.info(
             result['networkInterfaces'][0]['accessConfigs'][0]['natIP']
         )
 
-    def snapshot(self, instance, clean=True):
+        return self.get_instance(
+            result['id'],
+            self.project,
+            self.zone,
+            name=result['name']
+        )
+
+    def snapshot(self, instance: GceInstance, clean=True, **kwargs):
```

Comment:
Actually, now that I see this again, we don't need the `**kwargs`...that's just for the base class in case any functions have more kwargs they can take (like oci). So no need to change ec2 either.

Reply:
+1 works for me